sklearn/utils/extmath.py

```diff
@@ -357,8 +364,9 @@ def randomized_svd(M, n_components, n_oversamples=10, n_iter=4,
 
     # Adjust n_iter. 7 was found a good compromise for PCA. See #5299
     if n_components < .1 * min(M.shape) and n_iter < 7:
-        warnings.warn("The number of power iterations is increased to 7 to "
-                      "achieve higher precision.")
+        if n_iter_specified:
+            warnings.warn("The number of power iterations is increased to "
+                      "7 to achieve higher precision.")
```

Comment:
wait why are we changing n_iter here if it was explicitly given? I don't understand the logic. We should do either a good default or we should do what the user said. If the user provided ``n_iter`` we should do what she said IMHO.